extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryKit.java

```diff
@@ -328,4 +321,29 @@ private ShuffleSpec findShuffleSpecForNextWindow(List<OperatorFactory> operatorF
 
     return new HashShuffleSpec(new ClusterBy(keyColsOfWindow, 0), maxWorkerCount);
   }
+
+  /**
+   * Override the shuffle spec of the last stage based on the shuffling required by the first window stage.
+   * @param queryId
+   * @param dataSourcePlan
+   * @param shuffleSpec
```

Comment:
nit: To confirm, this is the shuffleSpec for the stage that is going INTO the window function stage, not the shuffle spec after the window function stage, right? We should document that, or rename the variable.

Reply:
> nit: To confirm, this is the shuffleSpec for the stage that is going INTO the window function stage, not the shuffle spec after the window function stage, right?

Yes

> We should document that, or rename the variable.

I had added this in the javadoc: `Override the shuffle spec of the last stage based on the shuffling required by the first window stage.`